src/main/splitting/splitting.py

```diff
@@ -0,0 +1,106 @@
+import logging
+
+import pandas as pd
+
+from src.main.util import consts
+from src.main.preprocessing.code_tracker_handler import get_ct_language
+from src.main.splitting.tasks_tests_handler import check_tasks, create_in_and_out_dict
+from src.main.util.file_util import ct_file_condition, get_all_file_system_items, get_file_name_from_path,\
+    get_parent_folder_name, get_result_folder, write_based_on_language
+
+FRAGMENT = consts.CODE_TRACKER_COLUMN.FRAGMENT.value
+
+log = logging.getLogger(consts.LOGGER_NAME)
+
+
+def get_tasks_with_max_rate(tasks: list, test_results: list):
+    max_rate = max(test_results)
+    indices = [i for i, tr in enumerate(test_results) if tr == max_rate]
+    return [tasks[i] for i in indices], max_rate
+
+
+def check_tasks_on_correct_fragments(data: pd.DataFrame, tasks: list, in_and_out_files_dict: dict, file_log_info=""):
+    data[FRAGMENT] = data[FRAGMENT].fillna("")
+    # if run after preprocessing, this value can be taken from 'language' column
+    language = get_ct_language(data)
+    # add unique to logs
+    log.info(file_log_info + ", language is " + language + ", found " + str(data.shape[0]) + " fragments")
+
+    if language is not consts.LANGUAGE.NOT_DEFINED.value:
```

Comment:
It is difficult case when we have two NOT - is not and NOT DEFINED. Could you please swap if and else bodies?
`if language is consts.LANGUAGE.NOT_DEFINED.value: actions`
`else: actions`